ibis/backends/postgres/__init__.py

```diff
@@ -309,8 +345,31 @@ def list_tables(
         with self._safe_raw_sql(sql) as cur:
             out = cur.fetchall()
 
+        # Include temporary tables only if no database has been explicitly specified
+        # to avoid temp tables showing up in all calls to `list_tables`
+        if db == "public":
```

Comment:
Interesting, this is a bit of an edge case.

On the one hand, temporary tables are always visible to the user who created them. On the other, the intent of asking for tables in a specific schema would seem to preclude showing them.

What you've done here seems fine to me.